### ISSUE TITLE
[Request] Better document docker-compose solutions

### ISSUE THREAD

**Issue by accountForIssues** [standardnotes/standalone#2]
First off, thank you so much for developing an open source e2e encrypted solution for notes.

I would love to self host this but the current method is confusing and a lack of documentation means I can't write my own compose files (easily and without a lot of trial and error).

Moreover, someone who wants to self-host should fully know what's happening behind the scenes and not just depend on a simple script esp. when it concerns highly private data.

Some questions/suggestions off the top of my head:

#### Do I need to constantly check and pull the repo for updates ?

If one is using docker, chances are they are using something like watchtower to inform about and update their images. I would also prefer it that way.

I would much rather have a docker-compose yaml and a .env file. Everything else can and should be packed into an image.

#### There are several .sh files without documentation.

server.sh simply executes a couple docker-compose commands. I understand making self-host accessible and I assume with server.sh, you can add more functions. But if someone knows how to use docker, they are comfortable to execute a few extra commands.So, I'd suggest documenting what server.sh is doing and the commands it uses.

wait-for.sh and entrypoint.sh should be merged into the image since these require no interaction from the user. The host and port variables are already exported to the container environment using env files.

#### There are several .env files, often with repeating variables with no documentation.

eg., In the current way, a user has no way to know whether VAR_NAME in file1.env should be same or different to VAR_NAME in file2.env esp. when they don't know what VAR_NAME actually does. Hope you get my point.

So, a better way would be to separate the env files in something like this;

- **database.env** -> database host details, credentials: these are shared and are common to all the containers.
- **redis.env** -> redis details
- **secrets.env** -> jwt details and other secret keys

For all the service specific variables related to debug and datadog, it would be ideal to rename them as per service to avoid confusions. e.g. DATADOG_SERVER, DATADOG_AUTH and DATADOG_API_GATEWAY.

#### Why are we building an image ?

I can see that there is a standardnotes/syncing-server image on docker hub but we never use it, instead we pull this repo and build a full image off it. Can we not simply use that image or am I not understanding something ?

Building an image makes sense if it's somehow customized for each system/user but it doesn't seem like that's the case and the current instructions only ask the user to edit the .env files.

### To sum up

1. Clean up env files and document the variables.
2. Document the behind the scenes process of the services in the docker-compose file.

---

Nextcloud is much more complex service that also uses database, redis and worker services but all I need to self-host is a single docker-compose.yml and a single .env file. I hope I can do the same with Standard Notes some day.

**Comment by JaspalSuri** [comment]
Hi @accountForIssues, thank you for the feedback and suggestions; I've taken note of it for review. 🙂

**Comment by karolsojko** [comment]
> Since, this is a self host server, ideally it should have support SMTP server for sending e-mails to users. Is that not possible currently ? I don't see any reference to such values in the *.env files.

We are using an `email` service for that purpose to which we communicate via SNS/SQS pub-sub. At this point an alternative is not implemented.
